cmd/operator-opamp-bridge/operator/client.go

```diff
@@ -76,82 +76,134 @@ func NewClient(name string, log logr.Logger, c client.Client, componentsAllowed
 	}
 }
 
-func (c Client) labelSetContainsLabel(instance *v1alpha1.OpenTelemetryCollector, label, value string) bool {
-	if instance == nil || instance.GetLabels() == nil {
-		return false
+func (c Client) Apply(name string, namespace string, configmap *protobufs.AgentConfigFile) error {
+	c.log.Info("Received new config", "name", name, "namespace", namespace)
+
+	if len(configmap.Body) == 0 {
+		return errors.NewBadRequest("invalid config to apply: config is empty")
+	}
+
+	var collector v1beta1.OpenTelemetryCollector
+	err := yaml.Unmarshal(configmap.Body, &collector)
+	if err != nil {
+		return errors.NewBadRequest(fmt.Sprintf("failed to unmarshal config into v1beta1 API Version: %v", err))
```

Comment:
nit: we should just use fmt.Errorf here instead

Reply:
Can do. Can you tell me why though? I thought we wanted to return a `errors.NewBadRequest` if the user's input (in this case the config) isn't valid based on our (updated) rules

Reply:
it's usually when i see something that's just making an error that's a Sprintf it's a golangci lint rule to just use fmt.Errorf. You are right though that NewBadRequest is closer to what Kubernetes would actually return here, so let's just stick with it! :)